R/forecast.R

```diff
@@ -117,55 +95,89 @@ as_forecast.default <- function(data,
   if (!is.null(forecast_unit)) {
     data <- set_forecast_unit(data, forecast_unit)
   }
+  return(data)
+}
 
-  # assert forecast type is as expected
-  assert_forecast_type(data, desired = forecast_type)
-  forecast_type <- get_forecast_type(data)
 
-  # produce warning if old format is suspected
-  # old quantile format
-  if (forecast_type == "point" && "quantile" %in% colnames(data)) {
-    #nolint start: keyword_quote_linter
-    cli_warn(
-      c(
-        "Found column 'quantile' in the input data",
-        "i" = "This column name was used before scoringutils version 2.0.0.
-        Should the column be called 'quantile_level' instead?"
-      ),
-      .frequency = "once",
-      .frequency_id = "quantile_col_present"
-    )
-    #nolint end
-  }
-  #old binary format
-  if (forecast_type == "point") {
-    looks_binary <- check_input_binary(factor(data$observed), data$predicted)
-    if (isTRUE(looks_binary)) {
-      #nolint start: keyword_quote_linter duplicate_argument_linter
-      cli_warn(
-        c(
-          "All observations are either 1 or 0.",
-          "i" = "The forecast type was classified as 'point', but it looks like
-          it could be a binary forecast as well.",
-          "i" = "In case you want it to be a binary forecast, convert `observed`
-          to a factor. See `?as_forecast()` for more information."
-        ),
-        .frequency = "once",
-        .frequency_id = "looks_binary"
-      )
-      #nolint end
-    }
-  }
+#' @rdname as_forecast
+#' @export
+#' @importFrom cli cli_warn
+as_forecast_point <- function(data,
+                              forecast_unit = NULL,
+                              observed = NULL,
+                              predicted = NULL,
+                              model = NULL) {
+  data <- as_forecast_generic(data, forecast_unit, observed, predicted, model)
+  data <- new_forecast(data, "forecast_point")
```

Comment:
shouldn't they really have their own new_forecast_type function etc? I think this could be its own issue discussion though

Reply:
Current version is a bit shorter, your proposal is a bit cleaner 🤷. Happy to open an issue if you want. Otherwise I'd just circle back to it once/if we feel embarrassed again by the current solution :)

Reply:
Yes I think it needs its own issue.

Reply:
New issue is here: https://github.com/epiforecasts/scoringutils/issues/856